### PR TITLE
Add support for (?<name>expr) syntax

### PR DESCRIPTION
**Pull request by rschlussel-zz**
This is in addition to the already supported (?P<name>expr) syntax for
capturing groups

### CODE REVIEW

File: javatests/com/google/re2j/ParserTest.java [hunk 1]
```diff
@@ -223,6 +223,8 @@ public boolean applies(int r) {
 
     // Test named captures
     {"(?P<name>a)", "cap{name:lit{a}}"},
+    {"(?<name>a)", "cap{name:lit{a}}"},
+
```
Comment (sopel39) [inline javatests/com/google/re2j/ParserTest.java:227]: why additional empty line?

**Comment by sopel39** [comment]
lgtm, but you could also record group names with their index and store them in Prog instance. We need in `appendReplacement` function.

**Comment by sopel39** [comment]
I will do group names storing, please push this commit

File: javatests/com/google/re2j/ParserTest.java (hunk 1 above)
Comment (rschlussel-zz) [inline javatests/com/google/re2j/ParserTest.java:227]: unintentional white space change.  I'll revert.

**Comment by rschlussel-zz** [comment]
merged

**Pull request closed without merging**